jib-cli/src/main/java/com/google/cloud/tools/jib/cli/Jar.java

```diff
@@ -195,15 +114,17 @@ public Integer call() {
                 + jarFile);
         return 1;
       }
-      if (!entrypoint.isEmpty() && !jvmFlags.isEmpty()) {
+      if (!commonContainerConfigCliOptions.getEntrypoint().isEmpty() && !jvmFlags.isEmpty()) {
         logger.log(LogEvent.Level.WARN, "--jvm-flags is ignored when --entrypoint is specified");
       }
 
       CacheDirectories cacheDirectories =
           CacheDirectories.from(commonCliOptions, jarFile.toAbsolutePath().getParent());
-      ArtifactProcessor processor = ArtifactProcessors.from(jarFile, cacheDirectories, this);
+      ArtifactProcessor processor =
+          ArtifactProcessors.from(jarFile, cacheDirectories, this, commonContainerConfigCliOptions);
       JibContainerBuilder containerBuilder =
-          JarFiles.toJibContainerBuilder(processor, this, commonCliOptions, logger);
+          JarFiles.toJibContainerBuilder(
+              processor, this, commonCliOptions, commonContainerConfigCliOptions, logger);
```

Comment:
Just got this idea: both `commonCliOptions` and `commonContainerConfigCliOptions` are a field of `this` (the `Jar` class). Maybe `Jar` can have getters for that, and then you only need to pass `this`?

But not sure if this is a bad or good idea.

Reply:
Ah that is an interesting idea and would certainly help us avoid passing in another parameter. But when `War.class` is implemented, it would also have to include those same getters so I don't know if we want duplicate that code?

Reply:
I think getters will be 1-liner. But then, I just realized `War` and `Jar` should probably implement a common interface if we want others to call those getters. Maybe let's just go with the current approach?